src/net/peer/connect/connect.rs

```diff
@@ -81,170 +83,161 @@ quick_error! {
         TimedOut {
             description("connection attempt timed out")
         }
+        DeadChannel {
+            description("Communication channel was cancelled")
+        }
+        RendezvousConnect {
```

Comment:
This should still carry the error from where it's used.

Reply:
Hmm, how do I do that?
Sometimes it's https://docs.rs/future-utils/0.5.0/future_utils/mpsc/struct.SendError.html and sometimes it's https://docs.rs/futures/0.1.17/futures/sync/oneshot/struct.Canceled.html

But they both basically mean the same thing: the other side of synchronization channel was dropped.

Reply:
I can only find `RendezvousConnect` being created in one place: https://github.com/maidsafe/crust/pull/887/files#diff-56c4cb5be4a0c20c65408095750c183eR278

Reply:
oh, I though you were about `DeadChannel` error. Never mind then :)

Reply:
Hmm, `RendezvousConnectError` is generic this applies https://github.com/tailhook/quick-error/issues/20
Currently, I changed it to 
`  RendezvousConnect(e: TcpRendezvousConnectError<UnboundedBiChannel<Bytes>>) {`

but that looks quite fragile, doesn't it?